lib/project_types/script/layers/domain/metadata.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Script
+  module Layers
+    module Domain
+      class Metadata
+        attr_reader :schema_major_version, :schema_minor_version
+
+        def initialize(schema_major_version, schema_minor_version)
+          @schema_major_version = schema_major_version
+          @schema_minor_version = schema_minor_version
+        end
+
+        class << self
+          def create_from_json(metadata_json)
+            metadata_hash = JSON.parse(metadata_json)
+            schema_versions = metadata_hash["schemaVersions"]
+            if schema_versions.nil?
+              err_msg = "Metadata is missing schemaVersions"
+              raise ::Script::Layers::Domain::Errors::MetadataValidationError, err_msg
+            end
+            # Scripts may be attached to more than one EP in the future but not right now
+            unless schema_versions.count == 1
+              err_msg = "Metadata schemaVersions should have one key"
+              raise ::Script::Layers::Domain::Errors::MetadataValidationError, err_msg
+            end
+
+            _, version = schema_versions.first
+            schema_major_version = version["major"]
+            schema_minor_version = version["minor"]
+            if schema_major_version.nil?
+              err_msg = "Metadata schema version is missing major key"
+              raise ::Script::Layers::Domain::Errors::MetadataValidationError, err_msg
+            end
+            is_prerelease = schema_major_version == "prerelease"
+
+            if schema_minor_version.nil? && !is_prerelease
```

Comment:
Giving this more thought, I'm thinking this check probably doesn't belong in the CLI. Technically this case is actually valid, it's just something extension point developers _shouldn't_ do until msgpack is more rolled out because not sending the minor version may result in the wrong schema being assigned (which won't matter in a msgpack world).

Reply:
removed